### PR TITLE
Fixes #493 #111

### PR DESCRIPTION
**Pull request by uilianries**
Related issue: https://github.com/bincrafters/community/issues/493
Related issue: https://github.com/bincrafters/community/issues/111

NOTE:
- There are no options for `no_main` and `debug_postfix` as mentioned on #111, thus I created virtual options.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

**Comment by SSE4** [comment]
looks good
@uilianries if you're already looking at gtest, may be you can also solve https://github.com/bincrafters/community/issues/111 ?

**Comment by uilianries** [comment]
Sure! Checking now!

**Comment by uilianries** [comment]
@SSE4 Could you review again, please?

File: test_package/CMakeLists.txt [hunk 1]
```diff
@@ -1,17 +1,31 @@
-project(test_package)
-cmake_minimum_required(VERSION 2.8.11)
+project(test_package CXX)
+cmake_minimum_required(VERSION 3.1.2)
```
Comment (SSE4) [inline test_package/CMakeLists.txt:1]: this must go first

File: conanfile.py [hunk 2]
```diff
@@ -11,66 +13,101 @@ class GTestConan(ConanFile):
...
     generators = "cmake"
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "build_gmock": [True, False], "fPIC": [True, False]}
-    default_options = ("shared=False", "build_gmock=True", "fPIC=True")
+    options = {"shared": [True, False], "build_gmock": [True, False], "fPIC": [True, False], "no_main": [True, False], "debug_postfix": "ANY"}
+    default_options = {"shared": False, "build_gmock": True, "fPIC": True, "no_main": False, "debug_postfix": 'd'}
+    _source_subfolder = "source_subfolder"
+
+    @property
+    def _suffix(self):
```
Comment (SSE4) [inline conanfile.py]: why not _postfix?

**Review by SSE4 — APPROVED** [review]

File: test_package/CMakeLists.txt (hunk 1 above)
Comment (uilianries) [reply test_package/CMakeLists.txt:1]: Done!

File: conanfile.py (hunk 2 above)
Comment (uilianries) [reply conanfile.py]: Done!

**Pull request merged**